toplevel/opttopmain.ml

```diff
@@ -20,6 +20,26 @@ let usage =
 
 let preload_objects = ref []
 
+(* Position of the current last expanded argument *)
+let last_expand_pos: int option ref = ref None
+
+let current = ref (!Arg.current)
+
+let argv = ref Sys.argv
+
+(* Test whether the option is part of a responsefile *)
+let is_expanded pos =
+  match !last_expand_pos with
+  | None -> false
+  | Some epos ->  pos <= epos
+
+let expand_position pos len =
+  let pos = match !last_expand_pos with
+    | Some opos when pos < opos ->
+        opos+len (* Already expand option just shift the position *)
+    | _ -> pos + len + 1 in (* New last position *)
```

Comment:
Should that not be `pos + len - 1` instead? I'm not sure what `len + 1` means in this context.

I think that you could store the outer limit of all expansions, that is the first position that is *not* part of an expansion (that would be `pos + len`, if I understand correctly). In that case, you would not need to use a `int option`, because you could initialize `first_nonexpanded_pos` with `ref 0`, and the code would be even more straightforward.